iam.tf

```diff
@@ -7,7 +7,7 @@ data "aws_iam_policy_document" "assume_role" {
 
     principals {
       type        = "Service"
-      identifiers = ["${slice(list("lambda.amazonaws.com", "edgelambda.amazonaws.com"), 0, var.lambda_at_edge ? 2 : 1)}"]
+      identifiers = slice(list("lambda.amazonaws.com", "edgelambda.amazonaws.com"), 0, var.lambda_at_edge ? 2 : 1)
```

Comment:
these are also backward incompatible, although it's possibly doable. I just didn't bother to work on this for now because if the previous changes turn out to be really incompatible, then there's no point in putting the effort for these ones